core-bundle/src/Resources/contao/pages/PageRegular.php

```diff
@@ -692,39 +714,26 @@ protected function createFooterScripts($objLayout, $objPage = null)
 			}
 		}
 
-		// Add search index metadata
-		if ($objPage !== null)
-		{
-			$noSearch = (bool) $objPage->noSearch;
-
-			// Do not search the page if the query has a key that is in TL_NOINDEX_KEYS
-			if (preg_grep('/^(' . implode('|', $GLOBALS['TL_NOINDEX_KEYS']) . ')$/', array_keys($_GET)))
-			{
-				$noSearch = true;
-			}
-
-			$meta = array
-			(
-				'@context' => array('contao' => 'https://schema.contao.org/'),
-				'@type' => 'contao:Page',
-				'contao:title' => $objPage->pageTitle ?: $objPage->title,
-				'contao:pageId' => (int) $objPage->id,
-				'contao:noSearch' => $noSearch,
-				'contao:protected' => (bool) $objPage->protected,
-				'contao:groups' => array_map('intval', array_filter((array) $objPage->groups)),
-				'contao:fePreview' => System::getContainer()->get('contao.security.token_checker')->isPreviewMode()
-			);
-
-			$strScripts .= '<script type="application/ld+json">' . json_encode($meta) . '</script>';
-		}
-
 		// Add the custom JavaScript
 		if ($objLayout->script)
 		{
 			$strScripts .= "\n" . trim($objLayout->script) . "\n";
 		}
 
 		$this->Template->mootools = $strScripts;
+
+		$this->Template->jsonLdScripts = function ()
```

Comment:
/cc @ausi?

Reply:
> this will not work if someone adds JSONLD data in the fe_page (or any included) templates.

Why? The callback function is only executed at the point where it gets inserted into the template and as this is just before `</body>` all other included template code would already have been executed.

> It will also be a problem when insert tags are being replaced on the response…

Same with page title or everything else so IMO a known limitation.

Reply:
we use `[[TL_BODY]]` and `[[TL_MOOTOOLS]]` etc. to support this, because otherwise e.g. a `{{insert_content::X}}` could not add global scripts. I suggest we use a `[[TL_JSONLD]]` to achieve the same? Maybe keep state in the class whether the closure was called at all and JSONLD should be replaced?

Reply:
In which case we're back to `str_replace` and could also have a general `FinalizeResponseContextEvent` instead of `JsonLdEvent` which we explicitly did not want.

Reply:
I don't think thats the same. We would stick to existing implementation details, but the template decides where and if the JSON is added.

Reply:
But thats ultimately up 2 u. I can guarantee this will come up pretty quickly and will require a fix, but the fix would still be easy and we can merge as is 😊